### PR TITLE
37.5 methodology state machine

### PR DESCRIPTION
**Pull request by AnnaGsell**
Here is an explanation of the state diagram, which is short because more details would probably collide with the implementation and user journey section.
Please check my English and the image embedding... Thanks

### CODE REVIEW

File: chapters/methodology.tex [hunk 1]
```diff
@@ -1,4 +1,4 @@
-\section{Methodology} \label{sec:methodology}
+\chapter{Methodology} \label{sec:methodology}
```
Comment (craftamap) [inline chapters/methodology.tex]: why?

File: chapters/methodology.tex [hunk 2]
```diff
@@ -9,8 +9,24 @@ \subsection{Architecture}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
 \subsection{State Machine}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+\usepackage{graphicx}
```
Comment (craftamap) [inline chapters/methodology.tex]: no imports inline, please. Move it into report.tex

File: chapters/methodology.tex [hunk 3]
```diff
@@ -9,8 +9,24 @@ \subsection{Architecture}
...
+
+
+\begin{figure}
+  \includegraphics[width=\linewidth]{../figures/state_diagramm.jpg}
+  \caption{Figure 1: The state diagramm.}
+  \label{fig:state_diagramm}
+\end{figure}
+
+
+A state chnage can be invoked throw three different kind of actions: An user action (pressing a button), a timeout after a specified couple of secondsm or an significant analyser result (faced detection or smile detection).
```
Comment (craftamap) [inline chapters/methodology.tex]: chnage

Comment (craftamap) [inline chapters/methodology.tex]: I'd suggest writing "as example pressing a button"

Comment (craftamap) [inline chapters/methodology.tex]: secondsm

Comment (craftamap) [inline chapters/methodology.tex]: faced

File: chapters/methodology.tex [hunk 4]
```diff
@@ -9,8 +9,24 @@ \subsection{Architecture}
...
+Note that if a face is detected and the user moves his face out of the frame afterwards, the process continues and handles this event indirectly as missing smile.
+Consistently, the face leaving the camera frame after a first smile detection is not dealed with explicitly but leads to low smile results.
+Besides there is no end knot, because like most smartphone applications the flow always returns to the start screen, represented in the start state.
```
Comment (craftamap) [inline chapters/methodology.tex]: Besides is wrong here, as it's equal to "neben" / "nebst".

File: chapters/methodology.tex [hunk 5]
```diff
@@ -9,8 +9,24 @@ \subsection{Architecture}
...
 
 \subsection{User Interface}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
+sss
```
Comment (craftamap) [inline chapters/methodology.tex]: Whats that?

File: chapters/background.tex [hunk 6]
```diff
@@ -1,4 +1,13 @@
 \section{Background} \label{sec:background}
 % identifying existing approaches within the research literature; this may potentially include mobile-based and non-mobile approaches if necessary
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren.
+Since Viola and Jones proposed "Rapid Object Detection using a Boosted Cascade of Simple Features" in 2001, their concept is the basis of most face detection approaches.
```
Comment (craftamap) [inline chapters/background.tex]: This is another pr, right?

**Review by craftamap — CHANGES_REQUESTED** [review]

File: chapters/background.tex [hunk 7]
```diff
@@ -1,4 +1,13 @@
...
+It's an extremely fast and solid machine learning approach for object detection in videos and pictures, e. g. faces.
+
+In detail, the concept is distinguished by three key components.
+The first is the “Integral Image”, a new image representation which allows the detector to compute features very fast.
+Secondly an AdaBoost based learning algorithm selects a small number of promising visual features and constructs very efficient classifiers.
```
Comment (limkokyew) [inline chapters/background.tex]: I think a comma is missing here: "Secondly, an AdaBoost [...]"

File: chapters/methodology.tex [hunk 8]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+\begin{figure}
+  \includegraphics[width=\linewidth]{figures/state_diagramm.jpg}
+  \caption{The state diagram.}
+  \label{fig:state_diagramm}
+\end{figure}
+
 
+Figure \ref{fig:state_diagramm} shows the states and transitions of the state machine.
+A state change can be invoked throw three different kind of actions: An user action (for example pressing a button), a timeout after a specified couple of seconds or an significant analyser result (face detection or smile detection).
```
Comment (limkokyew) [inline chapters/methodology.tex]: throw -> through
Maybe action -> interaction?
Maybe couple of seconds -> timeframe?

The "an" in "an significant analyser [...]" should be "a", I believe

File: chapters/methodology.tex [hunk 9]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
+
+Note that if a face is detected and the user moves his face out of the frame afterwards, the process continues and handles this event indirectly as missing smile.
```
Comment (limkokyew) [inline chapters/methodology.tex]: I'm not completely sure whether this is correct, but I commonly refer to a user as they (in order to include both genders)

File: chapters/methodology.tex [hunk 10]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
+Consistently, the face leaving the camera frame after a first smile detection is not dealed with explicitly but leads to low smile results.
```
Comment (limkokyew) [inline chapters/methodology.tex]: Not sure whether consistently is the right word here. Are you trying to emphasize the contrast between this statement and the previous one?

File: chapters/methodology.tex [hunk 11]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
+Furthermore instead of an end knot, the application flow returns to the start screen, represented in the start state.
```
Comment (limkokyew) [inline chapters/methodology.tex]: I think a comma is missing here: "Furthermore, instead of an [...]"

**Review by limkokyew — COMMENTED** [review]
Good work! There are some issues that I want to address though.

File: chapters/methodology.tex (hunk 10 above)
Comment (AnnaGsell) [reply chapters/methodology.tex]: I'm trying to show the simultaneity...

File: chapters/methodology.tex [hunk 12]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
+  \caption{The state diagram.}
+  \label{fig:state_diagramm}
+\end{figure}
+
 
+Figure \ref{fig:state_diagramm} shows the states and transitions of the state machine.
+A state change can be invoked through three different kind of actions: An user action (for example pressing a button), a timeout after a specified timeframe or a significant analyser result (face detection or smile detection).
+
+Note that if a face is detected and users move their faces out of the frame afterwards, the process continues and handles this event indirectly as missing smile.
+Consistently, the face leaving the camera frame after a first smile detection is not dealed with explicitly but leads to low smile results.
```
Comment (mac641) [inline chapters/methodology.tex]: Typo: dealed -> dealt

File: chapters/methodology.tex [hunk 13]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
+  \includegraphics[width=\linewidth]{figures/state_diagramm.jpg}
+  \caption{The state diagram.}
+  \label{fig:state_diagramm}
+\end{figure}
+
 
+Figure \ref{fig:state_diagramm} shows the states and transitions of the state machine.
+A state change can be invoked through three different kind of actions: An user action (for example pressing a button), a timeout after a specified timeframe or a significant analyser result (face detection or smile detection).
+
+Note that if a face is detected and users move their faces out of the frame afterwards, the process continues and handles this event indirectly as missing smile.
```
Comment (mac641) [inline chapters/methodology.tex]: I'd suggest using plural in the second part of the sentence as well when referring to not smiling, like `handles these events as missing smiles` or even simpler `as not smiling`

File: chapters/methodology.tex (hunk 10 above)
Comment (mac641) [reply chapters/methodology.tex]: Idk if this helps... If you like to point out the simultaneity, I'd use Moreover / Furthermore / in addition

File: chapters/methodology.tex [hunk 14]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
+Consistently, the face leaving the camera frame after a first smile detection is not dealed with explicitly but leads to low smile results.
+Furthermore, instead of an end knot, the application flow returns to the start screen, represented in the start state.
```
Comment (mac641) [inline chapters/methodology.tex]: - Is `Furthermore` really necessary in this context? I'd start with `Instead of having an end knot, the application...`. Though this decision depends on personal preferences...
- I'd use `node` instead of `knot`

**Review by mac641 — CHANGES_REQUESTED** [review]
Minor wording improvements/typos

**Review by diehmlukas — COMMENTED** [review]
If the requested changes are done, it's look good for me.

File: chapters/background.tex [hunk 15]
```diff
@@ -1,4 +1,13 @@
...
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren.
+Since Viola and Jones proposed ``Rapid Object Detection using a Boosted Cascade of Simple Features'' in 2001, their concept is the basis of most face detection approaches.
+It's an extremely fast and solid machine learning approach for object detection in videos and pictures, e. g. faces.
+
+In detail, the concept is distinguished by three key components.
+The first is the “Integral Image”, a new image representation which allows the detector to compute features very fast.
+Secondly, an AdaBoost based learning algorithm selects a small number of promising visual features and constructs very efficient classifiers.
+The last component is a method which combines increasingly more complex classifiers in a “cascade” where most of the computation time is spent on critical object-like parts of the given image by discarding non-promising regions early in the analysis process \cite{viola_jones}.
```
Comment (limkokyew) [inline chapters/background.tex]: Are the quotation marks around "cascade" functionally identical to the ones in "Rapid Object [...]"?

File: chapters/methodology.tex [hunk 16]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
+  \includegraphics[width=\linewidth]{figures/state_diagramm.jpg}
+  \caption{The state diagram.}
+  \label{fig:state_diagramm}
+\end{figure}
+
 
+Figure \ref{fig:state_diagramm} shows the states and transitions of the state machine.
+A state change can be invoked through three different kind of actions: An user action (for example pressing a button), a timeout after a specified timeframe or a significant analyser result (face detection or smile detection).
+
+Note that if a face is detected and users move their faces out of the frame afterwards, the process continues and handles these event indirectly as missing smiles.
```
Comment (limkokyew) [inline chapters/methodology.tex:31]: Face being singular in the first sentence and plural in the following one doesn't sound right to me. I'd rewrite the second sentence to also use singular forms.

"these event" -> "these events"

File: chapters/methodology.tex [hunk 17]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
+In Addition, the face leaving the camera frame after a first smile detection is not dealt with explicitly but leads to low smile results.
```
Comment (limkokyew) [inline chapters/methodology.tex]: "addition" without capital A

File: chapters/background.tex [hunk 18]
```diff
@@ -1,4 +1,13 @@
...
+It's an extremely fast and solid machine learning approach for object detection in videos and pictures, e. g. faces.
+
+In detail, the concept is distinguished by three key components.
+The first is the ``Integral Image'', a new image representation which allows the detector to compute features very fast.
+Secondly, an AdaBoost based learning algorithm selects a small number of promising visual features and constructs very efficient classifiers.
+The last component is a method which combines increasingly more complex classifiers in a ``cascade'' where most of the computation time is spent on critical object-like parts of the given image by discarding non-promising regions early in the analysis process \cite{viola_jones}.
+
+For that reason the approach is highly appropriate to fulfill the face detection task of the soHappy application.
+The open source computer vision and machine learning software library OpenCV provides a cascade classifier class for object detection, which uses the concept of Viola and Jones \cite{opencv_cascade_classifier}.
```
Comment (craftamap) [inline chapters/background.tex]: Wrong PR!

**Review by craftamap — CHANGES_REQUESTED** [review]

File: sources/sources.bib [hunk 19]
```diff
@@ -25,3 +25,17 @@ @article{sample_ref
...
+}
+
+@inproceedings{opencv_cascade_classifier,
+  author = {OpenCV team},
+  title = {Cascade Classifier},
+  year = {2020},
+  url = {https://docs.opencv.org/3.4/db/d28/tutorial_cascade_classifier.html},
+  note = {Retrieved September 16, 2020}, 
+}
+
```
Comment (craftamap) [inline sources/sources.bib]: This is propably also part of the other pr

**Review by craftamap — CHANGES_REQUESTED** [review]

**Review by mac641 — CHANGES_REQUESTED** [review]
Same as @craftamap mentioned

**Review by mac641 — APPROVED** [review]

File: chapters/methodology.tex [hunk 20]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
 
+Figure \ref{fig:state_diagramm} shows the states and transitions of the state machine.
+A state change can be invoked through three different kind of actions: An user action (for example pressing a button), a timeout after a specified timeframe or a significant analyser result (face detection or smile detection).
+
+Note that if a face is detected and users move their faces out of the frame afterwards, the process continues and handles these event indirectly as missing smiles.
+In addition, the face leaving the camera frame after a first smile detection is not dealt with explicitly but leads to low smile results.
+Instead of having an end node, the application flow returns to the start screen, represented in the start state.
+
+\subsection{User Interface}
+Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
```
Comment (craftamap) [inline chapters/methodology.tex]: This should also not be part of this pr.

File: chapters/methodology.tex [hunk 21]
```diff
@@ -9,8 +9,20 @@ \subsection{Architecture}
...
-\subsection{User Interface}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+\begin{figure}
+  \includegraphics[width=\linewidth]{figures/state_diagramm.jpg}
+  \caption{The state diagram.}
+  \label{fig:state_diagramm}
+\end{figure}
+
 
+Figure \ref{fig:state_diagramm} shows the states and transitions of the state machine.
```
Comment (craftamap) [inline chapters/methodology.tex]: Maybe elaborate state machine a bit more.
Whats a state, whats an action whats an transaction?

**Review by craftamap — CHANGES_REQUESTED** [review]

**Review by diehmlukas — APPROVED** [review]

**Review by craftamap — APPROVED** [review]
